### ISSUE TITLE
[epic][pose-generation] Improve candidate support quality and enable learned proposal policies

### ISSUE THREAD

**Issue by JanDuchscherer104** [JanDuchscherer104/ARIA-NBV#67]
## Outcome

Build a scientifically auditable candidate-generation stack that produces **diverse, physically plausible, target-relevant finite action sets**, exposes the knobs and evidence needed to tune those sets, and provides a clean migration path from handcrafted proposals to learned finite-candidate and continuous proposal policies.

This epic is a follow-up to PR #62. It is intentionally split into thesis-core data-quality work and post-core online/continuous-control work so that continuous learning does not destabilize the finite-candidate experiment before the latter is validated.

## Why this is needed

The current generator has a good structural contract—full-shell provenance, compact valid actions, hard invalidity masks, target-aware families, deterministic rollout replay, mesh rendering, and target-RRI labels—but the proposal distribution is still narrow and strongly coupled:

- component seeds are reused across rollout nodes;
- position and gaze families are coupled almost one-to-one;
- the core profile contains only forward-local, target-bearing, and lateral-bypass motions;
- target distance, target scale, selected-view history, remaining budget, and local scene clearance do not adapt the family schedule;
- free-space validity is approximated by an AABB plus point/segment clearance;
- configured resampling/refill does not yet provide a bounded final support contract;
- stored `sampler_probability = 1/N` is not the continuous proposal density;
- no presentation-free online oracle environment exists for learned behavior-policy collection;
- direct differentiation through the current hard renderer/RRI chain is neither required nor currently meaningful.

A negative planning result is interpretable only relative to the candidate support. The generator must therefore be treated as part of the experiment, not as incidental preprocessing.

## Core mathematical contract

At rollout step `t`, generation returns a full shell

```math
Q_t = \{q_{t,i}\}_{i=1}^{N_q}, \qquad
A_t = \{i : m_{t,i}=1\},
```

where `m` is a hard physical-feasibility mask. Invalid rows remain diagnostic/feasibility examples; they are not assigned artificial low RRI. Utility is defined only for oracle-label-supported valid rows.

Proposal quality and policy quality are separate quantities:

```math
\text{proposal regret}(Q)
= \max_{q\in Q_{ref}\cap A} r(q)
- \max_{q\in Q\cap A} r(q).
```

A learned selector must be compared on matched candidate tables. A learned proposer changes the action-support distribution and therefore requires separate proposal-quality, support, reachability, and behavior-policy provenance.

## Invariants across every child issue

1. **Actor/oracle separation.** Observed or predicted target descriptors may condition proposals. GT meshes, GT target crops, all-candidate renders, and target RRI remain oracle-only.
2. **Hard invalidity remains separate from utility.** A physically feasible low-gain view is retained as low-utility supervision. A physically invalid view is masked.
3. **Coordinate discipline.** Sampling uses explicit physical reference, camera, target, and gravity frames; display-only transforms cannot enter generation.
4. **Determinism is keyed, not globally reset.** Same immutable state key reproduces the same proposal table; a different root/history/replica receives a different stream.
5. **Matched comparisons.** Candidate-generator comparisons use an oracle selector; learned-selector comparisons freeze candidate generation.
6. **Resolved configuration is persisted per state.** Dynamic schedules must emit immutable resolved profiles, seeds, proposal lineage, and contract versions.
7. **Scene-disjoint evidence.** Roots, candidate replicas, and policy rollouts from one scene remain within one split and are aggregated at scene level.
8. **Hard-oracle evaluation remains final.** Surrogates may guide selection/refinement but cannot replace matched endpoint oracle re-evaluation.

### Phase A — improve data generation before broad regeneration

1. Fix deterministic random streams and add low-discrepancy/multi-root replicas.
2. Factor center and gaze generation; add missing high-value families.
3. Add state-aware schedules and an empirical Aria motion prior.
4. Add reservoir/refill/diversity selection and truthful proposal provenance.
5. Strengthen physical feasibility.
6. Freeze a proposal benchmark and select a canonical challenger profile.

### Phase B — learned finite-candidate control

7. Introduce a reusable online oracle environment with renderer/scorer caching.
8. Train and evaluate a learned selector over fixed candidate tables.

### Phase C — learned proposal generation

9. Learn family/parameter distributions and bounded residual proposals while retaining fixed anchors.
10. Add surrogate-guided differentiable refinement; keep direct soft-oracle differentiation an optional ablation.

## Existing dependencies and conflicts to resolve

- #54 already owns production detection of target-family collapse. The benchmark work in this epic must consume its evidence rather than duplicate its preflight owner.
- #55 owns truthful unavailable target diagnostics, required for target-framing and support analysis.
- #56 owns rejection of stale/legacy campaign evidence.
- PR #62 blocker fixes—especially target/GT-row identity, campaign split persistence, implementation-revision binding, and exclusive worker ownership—must land before regenerated artifacts are admitted.
- The draft thesis and #54 currently encode different root-support thresholds. Freeze that value in the resolved experiment contract before confirmatory generation; child issues must not silently change it.

## Definition of done

- [ ] A frozen reference-reservoir benchmark separates proposal regret from selector regret.
- [ ] A canonical 60-row profile has measured family survival, diversity, target framing, best-of-60 target gain, and resource cost on a scene-disjoint pilot.
- [ ] Candidate replicas and roots are deterministic, non-identical where intended, and fully provenance-bound.
- [ ] Physical validity includes endpoint and path-volume evidence rather than AABB membership alone.
- [ ] Every dynamic knob is either persisted as a model/input condition or frozen within one corpus.
- [ ] A learned selector can collect and train against the existing hard oracle without differentiating through it.
- [ ] A learned proposer logs exact behavior-policy/proposal provenance and retains fixed exploration anchors.
- [ ] Any differentiable refinement is validated by the original hard renderer and target-RRI oracle.
- [ ] Thesis-facing conclusions remain scoped to the frozen finite support and matched endpoint protocol.

## Internal pointers

- PR #62: https://github.com/JanDuchscherer104/ARIA-NBV/pull/62
- Candidate generation: `aria_nbv/aria_nbv/pose_generation/`
- Rollout engine: `aria_nbv/aria_nbv/rollouts/replay/engine.py`
- Renderer: `aria_nbv/aria_nbv/rendering/candidate_depth_renderer.py`
- Target oracle: `aria_nbv/aria_nbv/oracle/target_rri.py`
- Thesis: Sections 3.2.3–3.2.5, 4.2.4, 4.3, 5.2–5.3
- Seminar paper: Section XIII.B–C for the earlier multiple-subsequence, multiple-candidate-set, continuous-policy, and differentiable-refinement directions

## External orientation

- VIN-NBV — quality-driven candidate scoring: https://arxiv.org/abs/2505.06219
- GenNBV — learned 5-DoF free-space policy: https://openaccess.thecvf.com/content/CVPR2024/html/Chen_GenNBV_Generalizable_Next-Best-View_Policy_for_Active_3D_Reconstruction_CVPR_2024_paper.html
- Hestia — hierarchical continuous NBV and directional/face awareness: https://openaccess.thecvf.com/content/WACV2026/html/Lu_Hestia_Voxel-Face-Aware_Hierarchical_Next-Best-View_Acquisition_for_Efficient_3D_Reconstruction_WACV_2026_paper.html
- ObjView-Bench — support, budget, difficulty, and reachability: https://arxiv.org/abs/2605.10707
- PyTorch3D — renderer/operator substrate: https://arxiv.org/abs/2007.08501

## Non-goals

- Do not merge online continuous control into the thesis-core result before the finite-candidate headroom and learned-selector gates pass.
- Do not expose GT geometry or all-candidate renders to a V1 actor.
- Do not replace the hard oracle with a learned surrogate for final reporting.
- Do not create a generic simulator framework detached from the current typed candidate, replay, and oracle contracts.

**Comment by JanDuchscherer104** [comment]
## Research enrichment — round 2: move label fidelity and sampling design ahead of profile promotion

A second review of the thesis contract, Project Aria/ASE documentation, PyTorch3D camera semantics, reconstruction metrics, active-learning literature, and recent object-centric NBV work adds three missing work packages:

- **#79 — camera/depth/render/unprojection fidelity.** Project Aria documents fisheye camera models, while the current oracle adapts candidates to PyTorch3D `PerspectiveCameras`; ASE depth is documented as distance along the pixel ray. Logged-view render/depth replay and typed depth semantics are therefore a P0 gate.
- **#80 — target-RRI measurement fidelity.** The current face-average point–mesh metric and any-vertex OBB crop require duplicate-density, resolution, tessellation, crop, fusion-order, and near-solved normalization sensitivity before broad labels are interpreted.
- **#81 — population versus training acquisition.** Multi-root, replica, hard-mined, and policy-induced rows need a separate training acquisition manifest; the held-out population must remain outcome-independent and scene-disjoint.

### Revised order

1. Land the PR #62 identity/split/revision/worker-ownership blockers.
2. Pass #79 and select/freeze the oracle camera/depth contract.
3. Pass #80 and select/freeze the oracle crop/fusion/metric contract.
4. Implement #68–#72 candidate-support challengers.
5. Use #73 to select a candidate profile under matched oracle and attempt budgets.
6. Use #81 to scale training data without changing the confirmatory estimand.
7. Proceed to #74–#77 only after the offline support and metric gates pass.

### New cross-cutting invariants

- **A self-consistent render/unproject loop is not sufficient:** it must agree with logged ASE depth under the declared camera and depth semantics.
- **A larger point set is not automatically a better reconstruction:** exact duplicate evidence, raster resolution, and mesh tessellation must not create spurious RRI.
- **Final candidate width is not the only budget:** attempted proposals, renderer queries, oracle queries, and acquisition cost must be compared.
- **Family/proposal provenance can become a shortcut:** models need provenance-dropout and cross-profile tests.
- **Adaptive training acquisition is not population sampling:** every hard-mined, active, or policy-induced row needs selection provenance; primary evaluation stays on an untouched scene-level population.

### Additional primary/official sources

- Project Aria camera models: https://facebookresearch.github.io/projectaria_tools/docs/tech_insights/camera_intrinsic_models
- ASE depth contract: https://facebookresearch.github.io/projectaria_tools/docs/open_datasets/aria_synthetic_environments_dataset/data_format
- PyTorch3D cameras: https://pytorch3d.org/docs/cameras
- Density-aware Chamfer Distance: https://proceedings.neurips.cc/paper/2021/hash/0a49e3c3a03ebde64cfa4a1b4e0e4d0c-Abstract.html
- DAgger: https://proceedings.mlr.press/v15/ross11a.html
- Importance Weighted Active Learning: https://arxiv.org/abs/0812.4952
- Dataset Cartography: https://aclanthology.org/2020.emnlp-main.746/

These additions strengthen the epic's existing principle: proposal, feasibility, label, training-acquisition, and policy quality are distinct contracts and must be diagnosed separately.

**Comment by JanDuchscherer104** [comment]
### Additional missing target-data work package

Created **#82 — Freeze target-task admission, observed-to-GT association, uncertainty, and descriptor-perturbation contracts**.

This reconciles a material protocol difference: the current thesis draft describes GT-row-defined oracle target tasks, whereas PR #62 adds an observed-target association path. #82 prevents those from sharing ambiguous `matched`/row-index semantics, freezes the target population before association, preserves missed/ambiguous targets, and measures candidate/profile robustness to imperfect observed target geometry.

Revised P0 prerequisites are now:

```text
PR #62 identity/provenance fixes
→ #82 target identity/descriptor protocol
→ #79 camera/depth fidelity
→ #80 crop/fusion/RRI fidelity
→ candidate-support work #68–#73
```

#81 then owns enriched training acquisition without changing the frozen evaluation population.

**Comment by JanDuchscherer104** [comment]
## Implementation handoff — first wave

Recommended execution order for the next agents/PRs:

1. **PR #62 exact-head closeout** — restore green tests and re-verify correctness/provenance blockers on the current head. Do not rely on the stale exact-head validation currently stated in the PR body.
2. **#82 target-task / association contract** — first slice is typed identity separation + deterministic global observed→GT association + ambiguity/unmatched audit. Do **not** start descriptor-noise experiments yet.
3. **#79 camera/depth contract** — first slice is typed projection/depth semantics + pure conversion/round-trip tests; then logged-pose ASE mesh/depth replay evidence.
4. **#80 RRI robustness** — first slice is a read-only metamorphic/sensitivity harness. Do not replace the canonical metric until evidence demonstrates that a change is necessary.
5. **#68 candidate randomness/diversity** — first slice is state-keyed deterministic substreams + low-discrepancy sampling. Do not add new semantic pose families in this PR; those belong to #69.
6. **#73 benchmark scaffold** may be implemented in parallel as read-only infrastructure, but no candidate profile is promoted until #79/#80 are stable.

### Shared invariants
- Preserve the hard actor/oracle boundary and hard candidate-validity mask.
- Treat scene/target/action-support changes as versioned scientific-contract changes, not implementation details.
- Keep every patch single-purpose, typed, provenance-preserving, and covered by targeted regression tests.
- Any new scientific generation cohort waits until the upstream correctness, camera, and label-quality gates pass.
- #82 implementation can begin before #79/#80, but **scientific promotion/perturbation studies from #82 wait for #79/#80**.

Each child issue now has a concrete next-PR handoff comment with scope, tests, stop conditions, and definition of done.
